apps/jobs/views.py

```diff
@@ -322,6 +322,40 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
         )
 
 
+@api_view(["DELETE"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def challenge_submission_delete(request, challenge_pk, challenge_phase_pk, submission_pk):
+    """
+    Delete submission by the challenge Host
+    """
+    # check if the challenge exists or not
+    challenge = get_challenge_model(challenge_pk)
+
```

Comment:
You have to handle the exception being thrown if challenge doesn't exist and send a correct response for that right?

Reply:
@Ram81 If i have got it correctly then we are handling this exception in [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/base/utils.py#L75) so there shouldn't be any need to handle it here again. Please correct me if i am missing something here.